### PR TITLE
Improve invitation status display with expiry-aware logic

### PR DESCRIPTION
**Pull request by delano**
## Summary

Refactor invitation status handling to properly surface expired pending invitations and improve UI consistency across all invitation statuses.

### Changes

**Core Logic** (`src/types/organization.ts`):
- Added `LOCALIZED_INVITATION_STATUSES` set to define which statuses have localized labels (pending, accepted, declined, expired, revoked)
- Added `effectiveInvitationStatus()` function that converts pending invitations past their expiry to "expired" status, matching the countdown display
- Added `invitationStatusLabelKey()` function that maps statuses to i18n keys with graceful fallback for unknown/future statuses

**UI Updates** (`OrganizationSettings.vue`):
- Replaced hardcoded yellow "pending" badge with dynamic status-aware styling via `invitationStatusBadgeClass()`
- Added `invitationStatusLabel()` helper to resolve localized labels with fallback to raw status
- Now displays correct badge colors for all invitation statuses: green (accepted), red (declined), yellow (pending), gray (expired)

**Internationalization**:
- Added "revoked" status label to all 25 locale files for future-proofing (currently unused by API but label is wired for consistency)

### Benefits

- Pending invitations past expiry now correctly display as "expired" instead of misleading "pending"
- Consistent status badge styling across all invitation states
- Graceful degradation for unknown/future status values
- Comprehensive test coverage for status resolution logic

## Test Plan

Added comprehensive unit tests in `src/tests/types/organization-helpers.spec.ts` covering:
- Pending invitations with future/past expiry
- Expiry boundary conditions
- Non-pending statuses (accepted, declined, expired) remain unchanged
- Label key mapping for all localized statuses
- Fallback behavior for unknown statuses
- End-to-end status resolution (effective status + label mapping)

All tests pass with deterministic clock injection for reliable expiry comparisons.

https://claude.ai/code/session_013yCPRLjcUPBCvxGzx25eyk

### CODE REVIEW

File: src/apps/workspace/account/settings/OrganizationSettings.vue [hunk 1]
```diff
@@ -537,6 +538,28 @@ const formatTimeRemaining = (expiresAt: number): string => {
   }
 };
 
+/** Localized status label for an invitation, with raw-value fallback. */
+const invitationStatusLabel = (invitation: OrganizationInvitation): string => {
+  const status = effectiveInvitationStatus(invitation.status, invitation.expires_at);
```
[the remark this answers is not in the document]
Comment (delano) [reply src/apps/workspace/account/settings/OrganizationSettings.vue]: Fixed in 60bdff2b. The badge and countdown now derive from a reactive `useNow({ interval: 30_000 })` clock via a `decoratedInvitations` computed, so a pending invitation that lapses while the tab stays open flips to "Expired" on its own (within ~30s) instead of staying stale until the next action. Note the pre-existing countdown (`formatTimeRemaining`) had the same non-reactivity — it's now driven by the same clock, so the badge and the "expires soon" text stay in agreement.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/types/organization.ts [hunk 2]
```diff
@@ -106,3 +106,52 @@ import type { Organization } from '@/schemas/shapes/organizations/organization';
...
+export function effectiveInvitationStatus(
+  status: string,
+  expiresAtSeconds: number,
+  nowMs: number = Date.now()
+): string {
+  if (status === INVITATION_STATUSES.PENDING && expiresAtSeconds * 1000 <= nowMs) {
+    return INVITATION_STATUSES.EXPIRED;
+  }
+  return status;
+}
```
[the remark this answers is not in the document]
Comment (delano) [reply src/types/organization.ts:151]: Good catch — fixed in 60bdff2b. `effectiveInvitationStatus` now floors to whole seconds (`expiresAtSeconds <= Math.floor(nowMs / 1000)`), matching `formatTimeRemaining`'s `Math.floor(Date.now() / 1000)`. So the badge can no longer read "expired" while the countdown still says "expires soon" on a sub-second gap. Added two boundary tests: a sub-second-future expiry stays `pending`, and it flips to `expired` once the whole second ticks past.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by delano** [comment]
Thanks for the thorough review. Addressed the feedback in 60bdff2b:

**Stale expiry (also raised by Greptile P1)** — Fixed. Badge + countdown now derive from a reactive `useNow({ interval: 30_000 })` clock, so a pending invite that lapses while the tab is open updates on its own instead of showing a stale "Pending".

**Redundant double-compute of `effectiveInvitationStatus`** — Fixed. A single `decoratedInvitations` computed resolves the effective status, label, badge class, and countdown once per row; the template binds the pre-resolved fields.

**`revoked` badge switch asymmetry** — Made explicit: `case 'revoked'` now sits alongside `expired` (gray) with a comment, so the `switch` mirrors `LOCALIZED_INVITATION_STATUSES`.

**`pendingInvitationCount` counts raw `pending`, not effective status** — Intentional, now documented in-code. It must mirror the backend member-limit check (`create_invitation.rb`), which keeps a lapsed invitation's seat reserved until cleanup. Switching to the effective (expiry-aware) status would desync the UI quota check from the backend and let the UI attempt an invite the server rejects. So a row can read "Expired" in its badge while still counting toward the quota — deliberate.

**Copilot's ms-vs-whole-second point** — Fixed: `effectiveInvitationStatus` now floors to seconds to match `formatTimeRemaining`.

**Test coverage** — Added whole-second boundary tests (sub-second-future stays pending; flips once the second ticks past). The label/effective-status mapping and fallback path are covered by the pure-helper tests; the badge-class map is a pure switch over that same resolved status.

Local checks on the new commit: vitest 13/13, `vue-tsc` 0 errors, ESLint clean.

One point worth surfacing explicitly (from the earlier analysis, unchanged here): these accepted/declined/revoked branches are **latent today** — the list endpoint returns pending-only, declined records aren't indexed, and revoked invitations are deleted server-side. So in practice the badge renders Pending or Expired now; the rest is wired and ready for a future "invitation history" endpoint.

---
_Generated by [Claude Code](https://claude.ai/code)_
